homeassistant/helpers/template.py

```diff
@@ -2031,9 +2032,32 @@ def from_json(value):
 
 def to_json(value, ensure_ascii=True):
     """Convert an object to a JSON string."""
+    _LOGGER.warning("Template warning: 'to_json' is deprecated, use 'as_json' instead")
     return json.dumps(value, ensure_ascii=ensure_ascii)
 
 
+def as_json(value, pretty_print=False, sort_keys=False):
+    """Convert an object to a JSON string."""
+
+    def default(obj):
+        """Disable custom types."""
+        raise TypeError(f"Object of type {type(obj).__name__} is not JSON serializable")
+
```

Comment:
It looks like there are no closure variables here so we can move this outside the function

Reply:
Done.